rest-api/dao/participant_counts_over_time_service.py

```diff
@@ -176,70 +192,163 @@ def get_enrollment_status_sql(self, filters_sql_ps, filters_sql_p):
     # Add macros for GREATEST and LEAST, as they don't work in SQLite
     # Example: master/rest-api/dao/database_utils.py#L50
 
-    sql = """
-      SELECT
-         SUM(registered_cnt * (cnt_day <= calendar.day)) registered_participants,
-         SUM(member_cnt * (cnt_day <= calendar.day)) member_participants,
-         SUM(full_cnt * (cnt_day <= calendar.day)) full_participants,
-         calendar.day
-       FROM
-       (SELECT c2.day cnt_day,
-               registered.cnt registered_cnt,
-               member.cnt member_cnt,
-               full.cnt full_cnt
-            FROM calendar c2
-            LEFT OUTER JOIN
-            (SELECT COUNT(*) cnt,
-                (CASE WHEN (enrollment_status = 1 OR enrollment_status IS NULL) THEN
-                  DATE(p.sign_up_time)
-                  ELSE NULL END) day
+    sql_by_sample_status_time = """
+      select sum(is_register) as registered_participants,
+       sum(is_member) as member_participants,
+       sum(is_core) as full_participants,
+       z.day
+      from
+      (
+           select
+            x.participant_id,
+            case when register_day<=y.day and member_day>y.day then 1 else 0 end as is_register,
```

Comment:
So I believe that previously, registered included member, and member included full (as registered participants are a superset of members, and members a supserset of full participants.) I think we want to preserve that, which makes this logic incorrect?

Reply:
If a participant is already a member, we don't want to count it for registered, same for core, right? So in this SQL, if a participant is not a member, the member_day will be 3001-01-01, same for core_day.

Reply:
I just checked the dashboard UI, registered, member and core are separated.  Do you mean they separate them at the front end?